apps/aeoracle/src/aeo_query_tx.erl

```diff
@@ -215,20 +215,34 @@ for_client(#oracle_query_tx{sender        = SenderPubKey,
 %% -- Local functions  -------------------------------------------------------
 
 check_interaction(Q, Trees, Height) ->
-    Oracles = aec_trees:oracles(Trees),
-    Id      = aeo_interaction:id(aeo_interaction:new(Q, Height)),
-    case aeo_state_tree:lookup_interaction(Id, Oracles) of
+    Oracles  = aec_trees:oracles(Trees),
+    I        = aeo_interaction:new(Q, Height),
+    OracleId = aeo_interaction:oracle_address(I),
+    Id       = aeo_interaction:id(I),
+    case aeo_state_tree:lookup_interaction(OracleId, Id, Oracles) of
         none       -> ok;
         {value, _} -> {error, oracle_interaction_already_present}
     end.
 
-check_oracle(OraclePubKey, Trees, QueryFee) ->
+check_oracle(OraclePubKey, Trees, QueryFee, Height, QTTL, RTTL) ->
     OraclesTree  = aec_trees:oracles(Trees),
     case aeo_state_tree:lookup_oracle(OraclePubKey, OraclesTree) of
         {value, Oracle} ->
             case QueryFee >= aeo_oracles:query_fee(Oracle) of
-                true  -> ok;
+                true  -> check_oracle_ttl(Oracle, Height, QTTL, RTTL);
                 false -> {error, query_fee_too_low}
             end;
         none -> {error, oracle_does_not_exist}
     end.
+
+check_oracle_ttl(O, Height, QTTL, RTTL) ->
+    try
+        Delta  = aeo_utils:ttl_delta(Height, QTTL),
+        MaxTTL = aeo_utils:ttl_expiry(Height + Delta, RTTL),
+        case aeo_oracles:expires(O) < MaxTTL of
+            false -> ok;
```

Comment:
This checks that oracle interaction expires after (or at same time of) oracle. Ok.

Reply:
No, it checks that the oracle does not expire before the the oracle interaction. We do not want to risk orphaned oracle interactions.